### PR TITLE
style(mypy): Enforcing typing for a number of modules

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Adding type enforcement for a number of modules including: 

- queries
- security
- sql_validators
- tasks
 
### TEST PLAN

CI. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @dpgaspar @etr2460 @villebro

### CODE REVIEW

File: superset/connectors/druid/models.py [hunk 1]
```diff
@@ -823,6 +823,7 @@ def granularity(
 
         if origin:
             dttm = utils.parse_human_datetime(origin)
+            assert dttm
```
Comment (john-bodley) [inline superset/connectors/druid/models.py:826]: The `parse_human_datetime` allows for an optional `str` and thus returns `None`. This is somewhat of an anti-pattern (the check should occur before the function call) and should be fixed at some stage.

File: superset/tasks/schedules.py [hunk 2]
```diff
@@ -368,15 +391,17 @@ def schedule_email_report(
         schedule.id = schedule_id
         schedule.recipients = recipients
 
-    if report_type == ScheduleType.dashboard.value:
+    if report_type == ScheduleType.dashboard:
```
Comment (john-bodley) [inline superset/tasks/schedules.py:394]: This used to parse around the value (str) of the enum as opposed to the enum instance which seems somewhat atypical.

File: superset/utils/core.py [hunk 3]
```diff
@@ -687,25 +687,25 @@ def notify_user_about_perm_udate(granter, user, role, datasource, tpl_name, conf
...
+    mime_subtype: str = "mixed",
+) -> None:
     """
     Send an email with html content, eg:
     send_email_smtp(
         'test@example.com', 'foo', '<b>Foo</b> bar',['/dev/null'], dryrun=True)
     """
     smtp_mail_from = config["SMTP_MAIL_FROM"]
-    to = get_email_address_list(to)
+    to = get_email_address_list(to)  # type: ignore
```
Comment (john-bodley) [inline superset/utils/core.py:708]: Overriding of variable names of different types should be avoided in the future.

File: superset/utils/core.py [hunk 4]
```diff
@@ -1064,8 +1064,8 @@ def get_since_until(
             since, until = time_range.split(separator, 1)
             if since and since not in common_time_frames:
                 since = add_ago_to_since(since)
-            since = parse_human_datetime(since)
-            until = parse_human_datetime(until)
+            since = parse_human_datetime(since)  # type: ignore
```
Comment (john-bodley) [inline superset/utils/core.py:1067]: Overriding of variable names of different types should be avoided in the future.

Comment (dpgaspar) [reply superset/utils/core.py:1067]: Agree, we could use a different variable here

File: superset/security/manager.py [hunk 5]
```diff
@@ -580,7 +581,7 @@ def create_missing_perms(self) -> None:
             if pv.permission and pv.view_menu:
                 all_pvs.add((pv.permission.name, pv.view_menu.name))
 
-        def merge_pv(view_menu, perm):
+        def merge_pv(view_menu: str, perm: Model) -> None:
```
Comment (dpgaspar) [inline superset/security/manager.py:584]: `merge_pv(view_menu: str, perm: str) -> None:`

File: superset/utils/core.py (hunk 3 above)
Comment (dpgaspar) [reply superset/utils/core.py:708]: We can rename it to `smtp_mail_to` for example

**Review by dpgaspar — COMMENTED** [review]
Left a couple of comments

**Comment by john-bodley** [comment]
Thanks @dpgaspar for the review. I've addressed your comments.

File: superset/utils/core.py (hunk 3 above)
Comment (john-bodley) [reply superset/utils/core.py]: Done.

**Comment by john-bodley** [comment]
ping @dpgaspar and @villebro. I was hoping to get this PR merged to minimize the chance of conflicts.

File: superset/connectors/druid/models.py (hunk 1 above)
Comment (villebro) [reply superset/connectors/druid/models.py:826]: Agreed; was going to ask if we could do that change already now, but I expect this to be a Pandora's box, and probably warrants a separate PR to isolate regression risk.

**Review by villebro — APPROVED** [review]
LGTM. I think we should probably decide on a convention for `Enum` usage, eg. for an `Enum` variable `abc` when to do `abc.value` and when to just do `abc`. I've been burned a bunch of times by passing an `EnumClass(str, Enum)` to something that was expecting a `str`, only to have it do horrible side-effects, for example inserting the value `<Abc.XYZ: 'abc'>` into a database instead of `abc`. But we can debate this at a later date, I think this is good to go.

**Pull request merged**